### PR TITLE
feat-digital-clock-example

### PR DESCRIPTION
**Pull request by gabbapeople**
Update XOD case studies with the "digital clock guide".

### CODE REVIEW

**Comment by gabbapeople** [comment]
@evgenykochetkov @nkrkv @brusherru Guys, pls check out this PR.

File: docs/guide/rtc-example/README.md [hunk 1]
```diff
@@ -0,0 +1,333 @@
...
+- Fill the `YEAR`, `MON`, `DAY`, `HOUR`, `MIN`, and `SEC` pins with values. Each
+  pin contains a description of what form this values should be.
+
+For example, we set the current date. Now it is the 3rd of September of the 2018
+year, and the time is 14 hours 44 minutes and about 31 seconds. To set this date
+and time values to the RTC, we fill the `datetime` pins with values.
+
+- The `2018` year value to the `YEAR` pin.
+- `9` number of the September month to the `MON` pin.
+- `3` number of the day of the month to the `DAY` pin.
```
Comment (nkrkv) [inline docs/guide/rtc-example/README.md]: 17-th is used as a day below

File: docs/guide/rtc-example/README.md [hunk 2]
```diff
@@ -0,0 +1,333 @@
...
+entered.
+
+- `2018` number value at the `YEAR` pin.
+- `9` number value at the `MON` pin.
+- `17` number value at the `DAY` pin.
+- `14` number value at the `HOUR` pin.
+- `44` number value at the `MIN` pin
+- `31` number value at the `SEC` pin.
+
+Besides, the `datetime-unpack` node calculates the weekday and outputs it
```
Comment (nkrkv) [inline docs/guide/rtc-example/README.md]: `unpack-datetime`

File: docs/guide/rtc-example/README.md [hunk 3]
```diff
@@ -0,0 +1,333 @@
...
+through the `WD` pin in the range from `1` to `7`. Now, If we display this
```
Comment (nkrkv) [inline docs/guide/rtc-example/README.md]: :scroll: Now, if

File: docs/guide/rtc-example/README.md [hunk 4]
```diff
@@ -0,0 +1,333 @@
...
+values on the screen, we got the `2018917` date and `144431` time without any
+order and punctuation. That’s why we need to format this numbers to a proper
```
Comment (nkrkv) [inline docs/guide/rtc-example/README.md]: Hmm… not sure what do you mean by “if we display ~this~these values on the screen…” How can I do it? I see no easy way to see the mentioned “2018917” and “144431” values.

File: docs/guide/rtc-example/README.md [hunk 5]
```diff
@@ -0,0 +1,333 @@
...
+view before showing them at the display screen.  
+Different conventions exist all around the world for date and time
+representation, both written and spoken. Differences can exist in:
```
Comment (nkrkv) [inline docs/guide/rtc-example/README.md]: :scroll: for the date and time representation

File: docs/guide/rtc-example/README.md [hunk 6]
```diff
@@ -0,0 +1,333 @@
...
+[Unix time](https://en.wikipedia.org/wiki/Unix_time) formats. You can check out
+the `xod/datetime` library for the
+[`format-timestamp`](/libs/xod/datetime/format-timestamp/) node. It formats
+datetime for use in spreadsheet computer programms. In this example, let's
+format date and time as they are shown on the digital clock in the classic
+American style. Here is the picture of the digital clock we want to make.
+
+![Clock on the display screen](./text-lcd-16x2-i2c-with-time.png)
+
+Let’s create three additional nodes which format our date and time values.
```
Comment (nkrkv) [inline docs/guide/rtc-example/README.md]: nodes → patch nodes

File: docs/guide/rtc-example/README.md [hunk 7]
```diff
@@ -0,0 +1,333 @@
...
+
+### Formatting date
+
+We start the formatting process from the date. Create a patch for the first new
+patch node and name it `format-date`. We want this node to get a number of a
```
Comment (nkrkv) [inline docs/guide/rtc-example/README.md]: :scroll: to get numbers for a

File: docs/guide/rtc-example/README.md [hunk 8]
```diff
@@ -0,0 +1,333 @@
...
+year, month, and day and output them as a single string. Also, we want
+`format-date` to separate the date values with the `/` character.
+
+![Formatting date](./format-date.patch.png)
+
+- Add three `input-number` nodes onto the patch and name them `YEAR`, `MON`, and
+  `DAY`.
+- Add the `output-string` node onto the patch and name it `DATE`.
+- Add the [`format-number`](/libs/xod/core/format-number/) node and link it with
+  the `YEAR` node. The year values that comes to the `YEAR` pin are Number type.
```
Comment (nkrkv) [inline docs/guide/rtc-example/README.md]: :scroll: are of Number type

File: docs/guide/rtc-example/README.md [hunk 9]
```diff
@@ -0,0 +1,333 @@
...
+  The Number type values are floating point, and the incoming year value looks
+  like `2018.00`. We use the `format-number` node to get rid of those zeroes. If
+  you put the `0` value to the `DIG` pin string value of the year becomes
```
Comment (nkrkv) [inline docs/guide/rtc-example/README.md]: :scroll: ...to the `DIG` pin, the string value…

File: docs/guide/rtc-example/README.md [hunk 10]
```diff
@@ -0,0 +1,333 @@
...
+  The number of added zeroes depends on the width of the string. The width value
+  is set at the `W` input pin. This node is very useful for this guide. The
+  month and day value fields on our display have 2 digit capacity. If we show
+  the `12/22/2018` date on the screen, it is ok. However, if we show, for
+  example the 3rd of September directly without adding zeroes below the month
+  and day, the text on the screen looks will be `3/9/2018`. Thus cuts the width
+  of the date by two digits, and the total width of the line on the display
+  screen. The `pad-with-zeroes` solves this problem. We put the `2` width value
+  to the `W` pin on both nodes, so the 3rd of September looks like `03/09/2018`.
+  By this, the total width of the line with the date would always be static.
```
Comment (nkrkv) [inline docs/guide/rtc-example/README.md]: :scroll: would → will

File: docs/guide/rtc-example/README.md [hunk 11]
```diff
@@ -0,0 +1,333 @@
...
+  we make proper links between `join` input pins and other nodes. After this
+  link the `join` output to the `DATE` output-string node.
+
+### Formatting weekday
+
+The `datetime-unpack` node outputs the number of the weekday depending on the
+date. To show weekday names instead of numbers we need a node. Create a patch
+for this node and name it `format-weekday`.
+
+![Formatting weekday](./format-weekday.patch.png)
```
Comment (nkrkv) [inline docs/guide/rtc-example/README.md]: 300px?

File: docs/guide/rtc-example/README.md [hunk 12]
```diff
@@ -0,0 +1,333 @@
...
+  and year by a character of our choice. As we want to separate date values with
+  `/`, we link the `/` string constant with the `D` delimiter pin. With `join`
+  we can also set the order of the year, month, and day representation. As we
+  want the `MON` value to be first, `DAY` to be second, and `YEAR` to be third,
+  we make proper links between `join` input pins and other nodes. After this
+  link the `join` output to the `DATE` output-string node.
+
+### Formatting weekday
+
+The `datetime-unpack` node outputs the number of the weekday depending on the
```
Comment (nkrkv) [inline docs/guide/rtc-example/README.md]: `unpack-datetime`

File: docs/guide/rtc-example/README.md [hunk 13]
```diff
@@ -0,0 +1,333 @@
...
+  it outputs the selected value from `X` pins based on the `IDX` index value.
+  Let the input node `IN` be the index.
+- Link the `IN` input-node to the `IDX` pin.
+- Link the weekday nodes to the corresponding `X` pins of the `nth-input` node.
+
+### Formatting time
+
+Ok, we created nodes that format date and weekday. Time is left to format.
+Create a patch for the new node and name it `format-time`. This node will take
+hours, minutes and seconds from the `datetime-unpack` node and format them as we
```
Comment (nkrkv) [inline docs/guide/rtc-example/README.md]: :scroll: hours, minutes, and seconds

`unpack-datetime`

File: docs/guide/rtc-example/README.md [hunk 14]
```diff
@@ -0,0 +1,333 @@
...
+desire.
+
+![Formatting time step 1](./format-time-step-1.patch.png)
+
+- Add three `input-number` node for the time and name them `HOUR`, `MIN`, and
+  `SEC`.
+- Add the `output-string` node for the formatted time in a string form.
+- Put three `pad-with-zeroes` nodes onto the patch and link them with
+  `HOUR,`MIN`, and`SEC`nodes. To display hours, minutes, and seconds in a two-digit format, put the`2`value to the`W`pins as we do it in the`format-date`
```
Comment (nkrkv) [inline docs/guide/rtc-example/README.md]: Backtick pairing is messed up

File: docs/guide/rtc-example/README.md [hunk 15]
```diff
@@ -0,0 +1,333 @@
...
+- Hook-up wires
+
+<div class="ui segment note">
+<span class="ui ribbon label">Note</span>
+You can use a DS3231 or DS1302 breakout board Instead of DS1307 because they are interchangeable.
+</div>
+
+## Circuit
+
+![Circuit](./circuit.fz.png)
```
Comment (nkrkv) [inline docs/guide/rtc-example/README.md]: Should be no wider than 700px

File: docs/guide/rtc-example/README.md [hunk 16]
```diff
@@ -0,0 +1,333 @@
...
+  `am-pm` node.
+- Separate the time from the `PM` and `AM` labels with a single space. For this,
+  put another `join` node onto the patch. Put the spacer character to its
+  delimiter `D` pin.
+- Link the last `join` with the nodes above and the output node `TIME` to finish
+  the `format-time` patch.
+
+### Combining the formats
+
+Formats nodes are ready, so let’s put them onto the `rtc-example-read` patch.
```
Comment (nkrkv) [inline docs/guide/rtc-example/README.md]: :scroll: Format nodes are ready

File: docs/guide/rtc-example/README.md [hunk 17]
```diff
@@ -0,0 +1,333 @@
...
+
+![RTC read example step 2](./rtc-example-read.step2.patch.png)
```
Comment (nkrkv) [inline docs/guide/rtc-example/README.md]: 350px?

File: docs/guide/rtc-example/README.md [hunk 18]
```diff
@@ -0,0 +1,333 @@
...
+
+- Put the `format-date`, `format-weekday`, and `format-time` nodes, that you’ve
+  just created. Link their inputs to the `datetime-unpack` outputs.
```
Comment (nkrkv) [inline docs/guide/rtc-example/README.md]: `unpack-datetime`... Well, just use grep :) Will ignore the error for the rest of the article

File: docs/guide/rtc-example/README.md [hunk 19]
```diff
@@ -0,0 +1,333 @@
...
+- For the I2C 16x2 display, put the
+  [`text-lcd-16x2-i2c`](/libs/xod/common-hardware/text-lcd-16x2-i2c/) node onto
+  the patch.
+- The screen of the display in this example has 16x2 digits capacity. We align
+  both display lines to the center to make the clock looks better. For this, we
+  put two [`concat`](/libs/core/concat/) nodes for each line. There we added two
+  spaces to indent the text from the edge of the screen and one space to
+  separate the day of the week from the date.
+- Link the `concat` node of the date and time with the `L1` pin of the
+  `text-lcd-16x2-i2c` node. The second `concat` node with the `L2` pin.
```
Comment (nkrkv) [inline docs/guide/rtc-example/README.md]: :scroll: Link the second `concat`...

**Comment by gabbapeople** [comment]
@nkrkv Changes's been made.

File: docs/guide/rtc-example/README.md [hunk 20]
```diff
@@ -0,0 +1,362 @@
+---
+title: Digital clock example
+---
+
+# Digital clock example
```
Comment (nkrkv) [inline docs/guide/rtc-example/README.md]: Please, use the full title:

Digital Clock Example — Working with RTC Modules

Change:

1. Here
2. In the front matter `title`

File: docs/guide/rtc-example/README.md [hunk 21]
```diff
@@ -0,0 +1,362 @@
...
+  As we want to separate date values with `/`, we link the `/` string constant
+  with the `D` delimiter pin. With `join` we can also set the order of the year,
+  month, and day representation. As we want the `MON` value to be first, `DAY`
+  to be second, and `YEAR` to be third, we make proper links between `join`
+  input pins and other nodes. After this link the `join` output to the `DATE`
+  output-string node.
+
+### Formatting weekday
+
+The `datetime-unpack` node outputs the number of the weekday depending on the
```
Comment (nkrkv) [inline docs/guide/rtc-example/README.md]: `unpack-datetime`

File: docs/guide/rtc-example/README.md [hunk 22]
```diff
@@ -0,0 +1,362 @@
...
+`format-date` to separate the date values with the `/` character.
+
+![Formatting date](./format-date.patch.png)
+
+- Add three `input-number` nodes onto the patch and name them `YEAR`, `MON`, and
+  `DAY`.
+- Add the `output-string` node onto the patch and name it `DATE`.
+- Add the [`format-number`](https://xod.io/libs/xod/core/format-number/) node
+  and link it with the `YEAR` node. The year values that comes to the `YEAR` pin
+  are Number type. The Number type values are floating point, and the incoming
```
Comment (nkrkv) [inline docs/guide/rtc-example/README.md]: :scroll: are of `Number` type

**Comment by gabbapeople** [comment]
@nkrkv Updated! =)

**Comment by gabbapeople** [comment]
@nkrkv I removed the GIF. 
Suggest putting videos instead of GIFs at the `result` topics for all `case-studies`. Like it is made at `traffic-light`. And will be made at the `uart-communication` article.

**Pull request merged**